### ISSUE TITLE
Multiple GPU from library

### ISSUE THREAD

**Issue by mayaKaplansky** [OpenNMT/OpenNMT-tf#782]
Hi
I am trying to rum the model from the notebook using this command:
`runner.train(num_devices=1, with_eval=True)`
If I have 4 GPUs, should I change to` num_devices=4`? Will it use to train in parallel on all the GPUs and by that I can avoid OOM issues? Or is it just making the batches larger like explained in the website: 

> "OpenNMT-tf training can make use of multiple GPUs. In this mode, the graph is replicated over multiple devices and batches are processed in parallel. The resulting graph is equivalent to train with batches N times larger, where N is the number of used GPUs."?

**Comment by guillaumekln** [comment]
> If I have 4 GPUs, should I change to num_devices=4?

Yes.

> Will it use to train in parallel on all the GPUs and by that I can avoid OOM issues? Or is it just making the batches larger like explained in the website:

Each GPU will receive a batch a process it in parallel. Then the gradients are accumulated. You can search for "data parallelism".

**Comment by mayaKaplansky** [comment]
Hi I couldn't make it work and from another thread I understand that it will not work in Windows. Correct?

**Comment by guillaumekln** [comment]
I don't think multi-GPU training works on Windows. At this least it is not tested. I would recommend using a Linux system to get full support.

**Comment by mayaKaplansky** [comment]
Thank you, is there a way to define which GPU id the process should use? I want to run several models in parallel - how can I define that each model runs on a separate GPU (I have 4)?

**Comment by guillaumekln** [comment]
This is mentioned in the documentation: https://opennmt.net/OpenNMT-tf/faq.html#how-can-i-restrict-the-tensorflow-runtime-to-specific-gpu

**Comment by mayaKaplansky** [comment]
Thank you, that is for command line and I am using the library from a python notebook. Is there a way from the notebook?
These are my commands:

```

    runner = onmt.Runner(model, config,auto_config=True)
    print ('start training')
    output_dir, summary= runner.train(num_devices=1, with_eval=True, checkpoint_path=None, return_summary=True)
    print('end training')
    print(output_dir)
    print(summary)
    predictionFileName="PredictionsFull" + str(c)
    #inoutFiles=SrcTestFileName +"," + AgeGroup_test+"," + Gender_test, JobGroup_test, PatientLocation_test
    runner.infer(features_file=[SrcTestFileName, fileNameAgeGroupTest, fileNameGenderTest, fileNameJobGroupTest, fileNamePatientLocationTest], predictions_file=predictionFileName,log_time=True)

```

**Comment by mayaKaplansky** [comment]
Hi
I used 
```
import os
os.environ["CUDA_DEVICE_ORDER"]="PCI_BUS_ID"   # see issue #152
os.environ["CUDA_VISIBLE_DEVICES"]="0"
```
But when I run the code 4 times, each one with different  `CUDA_VISIBLE_DEVICES`, but they all crashed as if the data is mixed between the threads.
Any way to completely separate between them on the GPU?
